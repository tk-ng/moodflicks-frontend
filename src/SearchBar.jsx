import { useEffect, useState } from "react";
import {
	Form,
	FormGroup,
	Input,
	Button,
	InputGroup,
	Row,
	Col,
} from "reactstrap";
import { useSearchParams } from "react-router";

export const SearchBar = ({ search }) => {
	const [searchParams, setSearchParams] = useSearchParams();

	const keyword = searchParams.get("keyword") || "";
	const [formData, setFormData] = useState(keyword);

	useEffect(() => {
		setFormData(keyword);
	}, [keyword]);

	const handleChange = (evt) => {
		setFormData(evt.target.value);
	};

	const handleSubmit = async (evt) => {
		evt.preventDefault();
		if (formData.trim()) {
			setSearchParams({ keyword: formData });
			search(formData);
		}
	};
	console.log(search);
	console.log(search.name);
	const placeholder =
		search.name === "movieSearch"
			? "Search for a movie title..."
			: "Search by mood...";

	return (
		<Row className="mb-3">
			<Col xs="1"></Col>
			<Col>
				<Form onSubmit={handleSubmit} className="SearchBar">
					<FormGroup>
						<InputGroup>
							<Input
								name="search"
								placeholder={placeholder}
								value={formData}
								onChange={handleChange}
								required
							/>
							<Button>Search</Button>
						</InputGroup>
					</FormGroup>
				</Form>
			</Col>
			<Col xs="1"></Col>
		</Row>
	);
};
