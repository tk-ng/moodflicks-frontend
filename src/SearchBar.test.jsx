import { render, screen } from "@testing-library/react";
import { SearchBar } from "./SearchBar";
import { MemoryRouter, useSearchParams } from "react-router";
import userEvent from "@testing-library/user-event";
import { expect } from "vitest";

const search = vi.fn();

vi.mock("react-router", async function () {
	const mod = await vi.importActual("react-router");
	return {
		...mod,
		useSearchParams: vi.fn(),
	};
});

describe("Renders the SearchBar component", function () {
	it("mounts without crashing", async function () {
		useSearchParams.mockReturnValue([new Map(), () => vi.fn()]);
		const { container, asFragment } = render(
			<MemoryRouter>
				<SearchBar search={search} />
			</MemoryRouter>
		);
		expect(container).toBeInTheDocument();
		expect(asFragment).toMatchSnapshot();
	});

	it("submits with form data", async function () {
		const user = userEvent.setup();
		useSearchParams.mockReturnValue([new Map(), () => vi.fn()]);
		render(
			<MemoryRouter initialEntries={[""]}>
				<SearchBar search={search} />
			</MemoryRouter>
		);
		const searchBar = screen.getByPlaceholderText("Search", { exact: false });
		expect(searchBar).toBeInTheDocument();

		await user.type(searchBar, "testSearchBarInput");
		await user.click(screen.getByText("Search", { selector: "button" }));
		expect(search).toHaveBeenCalledWith("testSearchBarInput");
	});
});
