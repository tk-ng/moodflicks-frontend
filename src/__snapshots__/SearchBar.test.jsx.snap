// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders the SearchBar component > mounts without crashing 1`] = `[Function]`;
